Clients/GUI/Pages/InstrumentInfo.razor

```diff
@@ -0,0 +1,469 @@
+            }
+        }
+                },
+                Hover = new LineOptionsHover
+                {
+                    Intersect = true,
+                    Mode = InteractionMode.Y
+                },
+                Legend =
+    {
+        Display = false
+    }
+            }
+        };
+
+        tempDataSet = new LineDataset<TimeTuple<decimal>>
+        {
+            BorderColor = ColorUtil.FromDrawingColor(Color.CornflowerBlue),
+            Fill = false,
+            BorderWidth = 2,
+            PointRadius = 3,
+            PointBorderWidth = 1,
+            SteppedLine = SteppedLine.False
+        };
+    }
+
+    private void UpdateBalanceVariable()
+    {
+        currentBalance = currency switch
+        {
+            Currency.Rub => userBalance.BalanceInRub,
+            Currency.Usd => userBalance.BalanceInUsd,
+            Currency.Eur => userBalance.BalanceInEur
+        };
+    }
+
+    private void UpdateBalanceInCorrectCurrency()
+    {
+        switch (currency)
+        {
+            case Currency.Rub:
+                userBalance.BalanceInRub += cash;
+                break;
+            case Currency.Usd:
+                userBalance.BalanceInUsd += cash;
+                break;
+            case Currency.Eur:
+                userBalance.BalanceInEur += cash;
+                break;
+            default:
+                errorMessage = "This currency is disabled to trade";
+                errorOccured = true;
+                break;
+        }
+        cash = 0;
+    }
+
+    private async Task UpdateBalance()
+    {
+        oldBalance = (UserBalance)userBalance.Clone();
+        UpdateBalanceInCorrectCurrency();
+        var request = new UpdateUserBalanceRequest()
+        {
+            UserId = userBalance.UserId,
+            BalanceInRub = userBalance.BalanceInRub,
+            BalanceInUsd = userBalance.BalanceInUsd,
+            BalanceInEur = userBalance.BalanceInEur
+        };
+        try
+        {
+            var result = await BrokerManager.UpdateUserBalance(request);
+            if (result == true)
+            {
+                UpdateBalanceVariable();
+            }
+        }
+        catch
+        {
+            userBalance = oldBalance;
+            errorOccured = true;
+            errorMessage = "Can't update balance, try again later.";
+        }
+    }
+
+    private async Task GetCountFromPortfolio()
+    {
+        var instrumentInfo = await BrokerManager.GetInstrumentFromPortfolio(userBalance.UserId, instrument.Figi);
+        countInPortfolio = instrumentInfo.TotalCount;
+    }
+
+    private async Task GetUserBalance()
+    {
+        userBalance = await BrokerManager.GetUserBalance(userBalance.UserId);
+        UpdateBalanceVariable();
+    }
+
+    private async Task Sell()
+    {
+        UpdateBalanceVariable();
+        if (countInPortfolio < lotsToTrade * instrument.CountInLot)
+            Modal.Show<EmptyModal>("Count in portfolio less than you can sell");
```

Comment:
Do you really like avoiding brackets that much when there's one line of  `if`?

Reply:
It saves a place. Is it so bad?